.github/workflows/azure-e2e-test.yml

```diff
@@ -31,6 +31,17 @@ jobs:
         echo "Generated unique version for the image as: ${unique_version}"
         echo "image-version=${unique_version}" >> "$GITHUB_OUTPUT"
 
+  read-properties:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Read properties from versions.yaml
+      run: |
+        go_version="$(yq '.tools.golang' versions.yaml)"
+        [ -n "$go_version" ]
+        echo "GO_VERSION=${go_version}" >> "$GITHUB_ENV"
```

Comment:
Has this been tested? I think the env is just set for this job (`read-properties`), we need to use outputs if we want to carry variables from job to job, but why not just embed this in the build job?

Reply:
Actually this version is being used at multiple places it will be better to use  output rather than reading file multiple times.